### PR TITLE
Ei 1705 employment check error codes

### PR DESCRIPTION
**Pull request by David-B-Read**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/Map/DataExtensions.cs [hunk 1]
```diff
@@ -318,6 +320,7 @@ internal static Learner Map(this LearnerModel model)
                 LearningStoppedDate = model.SubmissionData.LearningData?.StoppedStatus?.DateStopped,
                 LearningResumedDate = model.SubmissionData.LearningData?.StoppedStatus?.DateResumed,
                 SuccessfulLearnerMatchExecution = model.SuccessfulLearnerMatch,
+                RefreshDate = model.RefreshDate
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/Map/DataExtensions.cs:323]: this is duplicated

File: src/SFA.DAS.EmployerIncentives.Database/Scripts/Generate.CompliancePausedEmploymentCheckFailuresReport.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+-- Failed 1 or Both
+SELECT x.AccountLegalEntityId, x.ULN, 'EmployedBeforeSchemeStarted' FROM
+(SELECT 
+    ai.Id,
+    ai.AccountLegalEntityId,
+	ai.ULN,
+	ai.PausePayments,
+	(SELECT Result FROM incentives.PendingPaymentValidationResult ppvrEC2 WHERE ppvrEC2.PendingPaymentId = pp.Id AND ppvrEC2.Step = 'EmployedBeforeSchemeStarted' AND ppvrEC2.PeriodNumber = @PeriodNumber AND ppvrEC2.PaymentYear = @PaymentYear) AS EmployedBeforeSchemeStarted,
+	(SELECT Result FROM incentives.PendingPaymentValidationResult ppvrLearn WHERE ppvrLearn.PendingPaymentId = pp.Id AND ppvrLearn.Step = 'HasLearningRecord' AND ppvrLearn.PeriodNumber = @PeriodNumber AND ppvrLearn.PaymentYear = @PaymentYear) AS HasLearningRecord
+FROM incentives.ApprenticeshipIncentive ai
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Database/Scripts/Generate.CompliancePausedEmploymentCheckFailuresReport.sql:22]: is this relevant to this feature?

File: src/SFA.DAS.EmployerIncentives.Database/Scripts/Generate.JetReport.sql [hunk 3]
```diff
@@ -26,6 +26,7 @@ SELECT
     ,case
         when ai.Phase = 'Phase1' then 1
         when ai.Phase = 'Phase2' then 2
+		when ai.Phase = 'Phase3' then 3
      end as [Phase]
     ,cast(ai.SubmittedDate as date) as [Application Date]
   FROM (
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Database/Scripts/Generate.JetReport.sql:32]: is this relevant?

**Review by ruslanss — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/Models/ApprenticeApplication.cs [hunk 4]
```diff
@@ -46,7 +46,11 @@ public class ApprenticeApplication
         public bool? InLearning { get; set; }
         public bool? FirstEmploymentCheckValidation { get; set; }
         public bool? FirstEmploymentCheckResult { get; set; }
+        [Column(TypeName = "nvarchar(50)")]
+        public EmploymentCheckResultError? FirstEmploymentCheckErrorType { get; set; }
         public bool? SecondEmploymentCheckValidation { get; set; }
         public bool? SecondEmploymentCheckResult { get; set; }
+        [Column(TypeName = "nvarchar(50)")]
+        public EmploymentCheckResultError? SecondEmploymentCheckErrorType { get; set; }
```
[the remark this answers is not in the document]
Comment (David-B-Read) [reply src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/Models/ApprenticeApplication.cs:54]: Duplicated with what? This is an EF model that's mapped to the incentives.ApprenticeApplications view

**Review by ruslanss — APPROVED** [review]

**Pull request merged**